src/mesh/coordinates.cxx

```diff
@@ -414,27 +414,16 @@ int Coordinates::calcCovariant() {
 
   free_matrix(a);
 
-  BoutReal maxerr, err;
-  maxerr = max(abs((g_11 * g11 + g_12 * g12 + g_13 * g13) - 1));
-  if ((err = max(abs((g_12 * g12 + g_22 * g22 + g_23 * g23) - 1))) > maxerr) {
-    maxerr = err;
-  }
-
-  if ((err = max(abs((g_13 * g13 + g_23 * g23 + g_33 * g33) - 1))) > maxerr) {
-    maxerr = err;
-  }
+  BoutReal maxerr;
+  maxerr= BOUTMAX(max(abs((g_11 * g11 + g_12 * g12 + g_13 * g13) - 1)),
+                  max(abs((g_12 * g12 + g_22 * g22 + g_23 * g23) - 1)),
+                  max(abs((g_13 * g13 + g_23 * g23 + g_33 * g33) - 1)));
 
   output_info.write("\tMaximum error in diagonal inversion is %e\n", maxerr);
```

Comment:
Shouldn't we do a global operation?
Otherwise this info can be misleading, if another processor has a larger error

Reply:
I've changed "Maximum error" to "Local maximum error" in the output